imageai/Prediction/Custom/custom_utils.py

```diff
@@ -24,11 +21,7 @@ def preprocess_input(x):
 
 def decode_predictions(preds, top=5, model_json=""):
 
-
-    global CLASS_INDEX
-
-    if CLASS_INDEX is None:
-        CLASS_INDEX = json.load(open(model_json))
+    CLASS_INDEX = json.load(open(model_json))
```

Comment:
I agree current implementation is not the best. However, this alternative isn't that better: each invocation on it will load exactly the same json.

If will let this change, should at leas add a with block to make sure the file is closed properly:

```python
with open(model_json) as f:
    CLASS_INDEX = json.load(f)
```
Agree on removing as much global variables as possible.

I don't fully understand why was on a global variable while the filename is a parameter... doesn't make sense for me, maybe I'm not seeing something.

Reply:
> I agree current implementation is not the best. However, this alternative isn't that better: each invocation on it will load exactly the same json.

Yeah but that's my point actually. With several instances of CustomImagePrediction running, each invocation will not load exactly the same json and the global variable CLASS_INDEX prevents us from running these instances at the same time.

> I don't fully understand why was on a global variable while the filename is a parameter... doesn't make sense for me, maybe I'm not seeing something.

This implementation is useful in other parts of codes in other "custom_utils.py" where some functions don't give a model_json in their parameters (but I checked, all functions that call this "custom_utils.py" give it).

PS : I may have misunderstood your concern, english is not my first language...

Reply:
My point is tat previous version loaded the file only once. We should go to an implementation with only one file-reading but avoiding the point you mention. A bigger refactor is needed, as I see.

English is not my first language too

thanks for your PR, we need to wait for @OlafenwaMoses now